cmake/FindRust.cmake

```diff
@@ -315,9 +315,25 @@ foreach(_VAR ${_Rust_USER_VARS})
     endif()
 endforeach()
 
+# If the user requested to use the rustup proxy binaries, just set the rustup directory we found as
+# the result
+if (DEFINED Rust_USE_RUSTUP_PROXY)
+    if (NOT Rust_RUSTUP)
+        set(_ERROR_MESSAGE "Rust_USE_RUSTUP_PROXY specified but rustup was not found.")
+        _findrust_failed(${_ERROR_MESSAGE})
+        return()
+    endif()
+    get_filename_component(_RUST_TOOLCHAIN_PATH "${Rust_RUSTUP}" DIRECTORY)
+    get_filename_component(_RUST_TOOLCHAIN_PATH "${_RUST_TOOLCHAIN_PATH}" DIRECTORY)
+    find_program(
+        Rust_COMPILER_CACHED
+        rustc
+            HINTS "${_RUST_TOOLCHAIN_PATH}/bin"
+            NO_DEFAULT_PATH)
+
 # Discover what toolchains are installed by rustup, if the discovered `rustc` is a proxy from
 # `rustup`, then select either the default toolchain, or the requested toolchain Rust_TOOLCHAIN
-if (_RESOLVE_RUSTUP_TOOLCHAINS)
+elseif (_RESOLVE_RUSTUP_TOOLCHAINS)
```

Comment:
How about if we expose this cache variable as a public "option". The behavior would be as follows:

- if rustup is not installed we FORCE the option to off.
- if rustup is installed we default the option to ON, but the user is allowed to configure the option to off. 

This was we only have one variable instead of two.

Reply:
Alright, I've changed it to be on top of the Rust_RESOLVE_RUSTUP_TOOLCHAINS (note that we still need a slightly different behavior when Rust_RESOLVE_RUSTUP_TOOLCHAINS is OFF for the case where rustup was found and rustc can be derived from its path, and the case where rustup was not found and rustc needs to be found independently).

Tell me if it looks okay and I'll squash the last commit into the other ones